serverless-lambda-container-otel-layer-example/Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM public.ecr.aws/lambda/nodejs:12
+
+COPY lambda-layer/ /opt
+
+COPY package.json .
+RUN npm i --production
```

Comment:
some node packages have native libraries that need to be compiled, and so require g++ and other tools. Will this docker image have the necessary tools?

Reply:
I don't think it does, I can do a multi-stage build example within this Dockerfile.

Reply:
Is that the same thing as:
1. run `npm i --production` outside the docker container
2. copy node_modules into the docker container
???

Reply:
correct, but if we do it within Dockerfile, it can guarantee it works fine in both local dev and prod since they are using the same instruction / version / deps.